include/dali/core/bitmask.h

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_CORE_BITMASK_H_
+#define DALI_CORE_BITMASK_H_
+
+#include <cassert>
+#include "dali/core/util.h"
+#include "dali/core/small_vector.h"
+
+namespace dali {
+
+/**
+ * @brief A vector of bits with a utility for quickly searching for set/cleared bits.
+ */
+class bitmask {
+ public:
+  using bit_storage_t = uint64_t;
+
+  static constexpr const int storage_bits = sizeof(bit_storage_t) * 8;
+  static constexpr const int storage_bits_log = ilog2(storage_bits);
+
+  /**
+   * @brief Find the first bit with given value, starting at start
+   *
+   * @return The index of the first bit found of the size of the bit mask if not found.
+   */
+  ptrdiff_t find(bool value, ptrdiff_t start = 0) const {
+    assert(start >= 0);
+    if (start >= size_)
+      return size_;
+    auto index = start >> storage_bits_log;
+    int bit = start & (storage_bits - 1);
```

Comment:
I wonder if moving `start & (storage_bits - 1)` to an utility function would make the code clearer (it repeats in multiple places)

Reply:
Perhaps.

Reply:
Done

Reply:
Similar for `something >> storage_bits_log`.